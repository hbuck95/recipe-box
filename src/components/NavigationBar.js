import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

import { library } from "@fortawesome/fontawesome-svg-core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUtensils } from "@fortawesome/free-solid-svg-icons";

import 'bootstrap/dist/css/bootstrap.css'

import RecipeForm from './RecipeForm';
import RecipeList from './RecipeList';
import RecipeView from './RecipeView';

import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
} from 'reactstrap';

library.add(faUtensils);

export default class NavigationBar extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isOpen: false
        };
    }

    toggle = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    render() {
        return (
            <Router>
                <div id="navbar">
                    <Navbar color="dark" light expand="md">
                        <NavbarBrand href="/">Recipe Box <FontAwesomeIcon icon="utensils" /></NavbarBrand>
                        <NavbarToggler onClick={this.toggle} />
                        <Collapse isOpen={this.state.isOpen} navbar>
                            <Nav className="ml-auto" navbar>
                                <NavItem>
                                    <Link id="addrecipe" to="/addrecipe">Add Recipe </Link>
                                </NavItem>
                                <NavItem>
                                    <br/>
                                </NavItem>
                                <NavItem>
                                    <Link to="/viewrecipes">View Recipes</Link>
                                </NavItem>
                            </Nav>
                        </Collapse>
                    </Navbar>

                    <Route exact path="/"/>
                    <Route path="/addrecipe" component={RecipeForm} />
                    <Route path="/viewrecipes" component={RecipeList} />
                    <Route path="/viewrecipe/:id" component={RecipeView}/>
                </div>
            </Router>
        );
    }
}