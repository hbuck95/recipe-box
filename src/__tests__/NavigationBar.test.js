import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import { shallow, configure } from 'enzyme';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { NavbarBrand } from 'reactstrap';

import NavigationBar from '../components/NavigationBar';

configure({adapter: new Adapter()});

it('Navigation bar renders', () => {
    const wrapper = shallow(<NavigationBar/>);
    const brand = <NavbarBrand href="/">Recipe Box <FontAwesomeIcon icon="utensils" /></NavbarBrand>
    expect(wrapper.contains(brand)).toEqual(true);
});

it('Navigation bar matches snapshot', () => {
    const wrapper = shallow(<NavigationBar/>);
    expect(wrapper).toMatchSnapshot();
});

it('Click "Create Recipe" button', () => {
    const wrapper = shallow(<NavigationBar/>);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('[id="addrecipe"]').simulate("click");
    expect(wrapper).toMatchSnapshot();

});