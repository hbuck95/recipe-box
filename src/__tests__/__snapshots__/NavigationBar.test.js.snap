// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click "Create Recipe" button 1`] = `ShallowWrapper {}`;

exports[`Click "Create Recipe" button 2`] = `ShallowWrapper {}`;

exports[`Click 'Create Recipe' button 1`] = `ShallowWrapper {}`;

exports[`Click 'Create Recipe' button 2`] = `ShallowWrapper {}`;

exports[`Navigation bar matches snapshot 1`] = `ShallowWrapper {}`;
